frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -1447,4 +1471,16 @@ public boolean existsLocalVersionFrameworkRequest() throws NotAvailableException
   protected ClusterConfiguration getClusterConfiguration() {
     return requestManager.getClusterConfiguration();
   }
+
+  protected Configuration getConfiguration() {
+    return this.conf;
+  }
+
+  protected StatusManager getStatusManager() {
+    return this.statusManager;
+  }
+
+  protected RequestManager getRequestManager() {
+    return this.requestManager;
+  }
```

Comment:
Do not expose them!

Reply:
Do not use it like am.conf, etc. They are designed to be private, and changed to protected only for Testing

---
In reply to: [174688494](https://github.com/Microsoft/pai/pull/318#discussion_r174688494) [](ancestors = 174688494)